src/components/Category.js

```diff
@@ -69,7 +78,22 @@ const CategoryComponent = ({ category, stateContext }) => {
       .equalTo(selectedState)
   }
   const [snapshots, loading, error] = useList(refToUse)
-  const dataSource = snapshots.map((i) => i.val())
+  const [searchedValue, setSearchedValue] = useState("")
+  const [filteredData, setFilteredData] = useState([])
+
+  useEffect(() => {
+    const dataSource = snapshots.map((i) => i.val())
```

Comment:
Can move this outside useEffect so it doesn't map again every time a user enters a key?

Reply:
The reason I moved in is because, the dataSource wasn't getting updated once snapshots were fetched. 
Will need to move the dataSource to state